### PR TITLE
Respec testing

### PR DESCRIPTION
**Pull request by Tresor11**

### CODE REVIEW

**Review by Redvanisation — CHANGES_REQUESTED** [review]
# Changes required  ♻️ 

### Hello team  🌟

**Nice work on this project, but we need to fix the issues below before moving on ⬇️** 

- You have covered both positive and negative scenarios for most cases which is great! however, you still didn't test all the methods, please test all methods you have in both scenarios positive and negative.

## [OPTIONAL]:
- Separate the code and put every class in a different file.

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Comment by WCanirinka** [comment]
Requesting for a code review

**Comment by Tresor11** [comment]
**we have added the test case of each of the method we used**
 and we separated our the files as suggested :smile:

File: spec/logic_spec.rb [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    run8 = Logic.new('P1', 'P2')
+    [1, 2, 5, 3, 9].each do |x|
+      run8.move(x)
+      next unless x == 9
+
+      it 'Checks win condition on first diagonal' do
+        expect(run8.game_end).to be_truthy
+      end
+    end
```
Comment (maelfosso) [inline spec/logic_spec.rb]: 1. Declare your variable **runX**
2. Usually it's preferable to loop when you want to expect directly without **next**. Morever, why use a loop for only 5 items

File: spec/logic_spec.rb [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    run9 = Logic.new('P1', 'P2')
+    [3, 2, 5, 1, 7].each do |x|
+      run9.move(x)
+      next unless x == 7
+
+      it 'Checks win condition on second diagonal' do
+        expect(run9.game_end).to be_truthy
+      end
+    end
+  end
```
Comment (maelfosso) [inline spec/logic_spec.rb]: Sorry, but it's a bad practise.
```ruby
decribe
  # declare let, subject....
  let (:run9) { Logic.new('P1', 'P2') }
  it '#check win condition' do
    run9.move(3)
    .....
    run9.move(7)
    expect(run9.game_end) # ....
  end
end
```
There is no need to write a loop for such situation

File: spec/logic_spec.rb [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    run = Logic.new('p1', 'p2')
+    1.upto(9) do |x|
+      it 'return true if the moves are correct' do
+        expect(run.move(x)).to eql(true)
+      end
+      it 'return false if the move are not available' do
+        expect(run.move(x + 9)).to eql(false)
+      end
+    end
+  end
```
Comment (maelfosso) [inline spec/logic_spec.rb]: Without the **next** it's OK.
Now declare the variable using **let**

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello coders,

Nice improvements :clap: But there are things to improve concerning your code quality.

1. Declare using `let` or `subject`. Please read this [Better specs](http://www.betterspecs.org/)

2. It's a bad practice to use loop like you're doing. Read the comments.

3. Think about using global variable.

Kindly make the changes and submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Comment by Tresor11** [comment]
thank you
**we have used global variables and removed the useless loops** :smile:

**Review by Redvanisation — CHANGES_REQUESTED** [review]
### Nicely done so far!

- There is a test case that is failing, please fix it all tests need to be green: 
```
  1) Logic#move return false if the move are not available
     Failure/Error: expect(runx.move(x * 9)).to be(false)
     
       expected false
            got true
```

### _After making the required changes please submit a new code review request_

### Good luck! 🍀

**Review by Redvanisation — APPROVED** [review]
# Project approved ✔️ 

### Hello team 🌟 

### _Nice work on this project, it seems that you have met the requirements so I am approving your project. 👍_ 

### You may merge to master and submit a project completion form. 🚀 

### Good luck! 🍀

**Pull request merged**